### ISSUE TITLE
Orphaned span not reported

### ISSUE THREAD

**Issue by tutufool** [openzipkin/brave#831]
Hi, 

I'm trying to upgrade my brave-opentracing to latest 0.33.6, but I found the orphaned span not reported.
Then I tried a lower version 0.33.0, it turns out to work. (brave 5.3.3)

The suspicious code is:
https://github.com/openzipkin/brave/blob/release-5.4.0/brave/src/main/java/brave/internal/recorder/PendingSpans.java#L107

according to:

  public final Boolean sampled() {
    return (flags & FLAG_SAMPLED_SET) == FLAG_SAMPLED_SET
        ? (flags & FLAG_SAMPLED) == FLAG_SAMPLED
        : null;
  }

shouldn't we apply FLAG_SAMPLED_SET as well?

**Comment by tutufool** [comment]
btw, here're my test case:

@Test
  public void testGCFinishForBrave() throws InterruptedException {
    final List<Span> buffer = new ArrayList<Span>();
    Reporter<Span> reporter = new Reporter<Span>() {

      @Override
      public void report(Span span) {
        buffer.add(span);

      }
    };

    Tracing brave4 = Tracing.newBuilder().localServiceName("TestServiceName").spanReporter(reporter)
        .currentTraceContext(CurrentTraceContext.Default.create()).build();

    BraveTracer tracer = BraveTracer.newBuilder(brave4).build();

    //do not call finish and try to see whether brave auto flush it by GC
    tracer.buildSpan("s1").start();
    tracer.buildSpan("s2").start();

    System.gc();

    Thread.sleep(5000);

    tracer.buildSpan("s3").start().finish();

    buffer.forEach(System.out::println);

    Assert.assertEquals(3, buffer.size());

    Assert.assertTrue(buffer.get(0).toString().contains("brave.flush"));
    Assert.assertTrue(buffer.get(1).toString().contains("brave.flush"));
    Assert.assertFalse(buffer.get(2).toString().contains("brave.flush"));

  }

**Comment by adriancole** [comment]
cc @devinsba

I dont see the obvious bug here but might with the test you gave. will look
into it when online next

**Comment by tutufool** [comment]
Yup, you can try with my test against brave-opentracing 0.33.6 (failed) and 0.33.0 (green)

My debug indicates that flags might be:

      TraceContext context = InternalPropagation.instance.newTraceContext(
          InternalPropagation.FLAG_SAMPLED | InternalPropagation.FLAG_SAMPLED_SET ,
          contextKey.traceIdHigh, contextKey.traceId,
          0L, contextKey.spanId,
          Collections.emptyList()
      );

**Comment by devinsba** [comment]
I don't see my symptom anywhere in the `brave-opentracing` codebase, though without stepping through with a debugger I can't be sure. It looks like there are hard references to all `TraceContext` objects. Just not sure if they are all singletons

**Comment by devinsba** [comment]
On further inspection I found only 1 possible culprit but that would result in an abandoned span instead of a flushed:

https://github.com/openzipkin-contrib/brave-opentracing/blob/master/src/main/java/brave/opentracing/BraveSpan.java#L91

I can't see any other way for a `TraceContext` to get copied or converted to a builder within this library

**Comment by adriancole** [comment]
@tutufool you are right! https://github.com/openzipkin/brave/pull/832

**Comment by adriancole** [comment]
5.5.1 on the way
